File: frontend/features/projects/types/project.ts
export type ProjectStatus = "draft" | "active" | "completed";

export interface Project {
  id: string;
  name: string;
  description: string | null;
  status: ProjectStatus;
  status_auto: boolean;
  brand_voice: string | null;
  created_at: string;
  updated_at: string;
}

export interface CreateProjectPayload {
  name: string;
  description?: string;
  brand_voice?: string;
}

export interface UpdateProjectPayload {
  name?: string;
  description?: string | null;
  status?: ProjectStatus;
  status_auto?: boolean;
  brand_voice?: string | null;
}

File: backend/app/database/models/project.py
from __future__ import annotations

from sqlalchemy import Boolean, ForeignKey, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database.base import Base
from app.database.mixins import BaseModelMixin


class Project(
    Base,
    BaseModelMixin,
):
    __tablename__ = "projects"

    owner_id: Mapped[str] = mapped_column(
        ForeignKey("users.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )

    name: Mapped[str] = mapped_column(
        String(255),
        nullable=False,
    )

    description: Mapped[str | None] = mapped_column(
        Text,
        nullable=True,
    )

    status: Mapped[str] = mapped_column(
        String(50),
        default="draft",
        nullable=False,
    )

    # When True, the system is free to move `status` forward automatically
    # based on project activity (e.g. draft -> active once content exists).
    # Set to False as soon as a user manually picks a status, so the system
    # never fights a deliberate choice.
    status_auto: Mapped[bool] = mapped_column(
        Boolean,
        default=True,
        nullable=False,
    )

    brand_voice: Mapped[str | None] = mapped_column(
        Text,
        nullable=True,
    )

    owner = relationship(
        "User",
        back_populates="projects",
    )
